src/wast-parser.cc

```diff
@@ -545,6 +570,15 @@ Result ResolveFuncTypes(Module* module, Errors* errors) {
     }
 
     if (func) {
+      if (!func->local_type_list.empty()) {
+        result |= ResolveRefTypes(*module, func->local_type_list,
+                                  func->GetNumParams(),
+                                  decl->sig.param_type_names, errors);
+
+        func->local_types.Set(func->local_type_list);
```

Comment:
This is a delayed processing of locals. The reasion is that otherwise we cannot tell that `(ref $t)` and `(ref 0)` refers to the same or different locals, so they can be merged or not.